General queries/Baseline Comparison.txt

```diff
@@ -0,0 +1,239 @@
+// Calculate the delta in network events, keeping the bad ones
+let NetworkDelta=GetNetworkEvents(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetNetworkEvents(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in process create events, keeping the bad ones
+let ProcessDelta=GetProcessCreates(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetProcessCreates(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in powershell events, keeping the bad ones
+let PSDelta=GetPSCommands(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetPSCommands(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in file create events, keeping the bad ones
+let FileDelta=GetFileCreates(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetFileCreates(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in logon events, keeping the bad ones
+let LogonDelta=GetLogonEvents(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetLogonEvents(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in registry events, keeping the bad ones
+let RegistryDelta=GetRegistryEvents(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetRegistryEvents(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in connected network events, keeping the bad ones
+let ConnectedNetworkDelta=GetConnectedNetworks(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetConnectedNetworks(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in image load events, keeping the bad ones
+let ImageLoadDelta=GetImageLoads(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetImageLoads(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Calculate the delta in raw IP address communications, keeping the bad ones
+let RawIPCommunicationDelta=GetRawIPCommunications(SuspectedBadHostMachineId, SuspectedBadLeftEventTime)
+| join kind=leftanti (
+    GetRawIPCommunications(GoodHostMachineId, GoodLeftEventTime)
+) on Entity;
+// Get the alerts for the bad machines (no delta, we care about all alerts)
+let Alerts=AlertEvents
+| where "Alert" in (ReturnSets)
+| where MachineId in (SuspectedBadHostMachineId)
+| summarize EventTime=max(EventTime), Count=count() by Title, MachineId, FileName, RemoteUrl
+| project EventTime, Entity=Title, Count, AdditionalData=coalesce(FileName, RemoteUrl), DataType="Alert";
+// String everything together
+union NetworkDelta, ProcessDelta, PSDelta, FileDelta, Alerts, LogonDelta, RegistryDelta,
+      ConnectedNetworkDelta, ImageLoadDelta, RawIPCommunicationDelta
+// Join back against the machine info so the Computer Names can be reassociated
+| join kind=leftouter (
```

Comment:
probably better to select ComputerName in the first place

Reply:
Updated in latest commit.